### PR TITLE
Create a metric card for Competition MATH

### PR DESCRIPTION
**Pull request by sashavor**
Proposing metric card for Competition MATH

### CODE REVIEW

File: metrics/competition_math/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+It first canonicalizes the inputs (e.g., converting `1/2` to `\\frac{1}{2}`) and then computes accuracy.
+
+## How to use 
+
+This metric takes two arguments:
+
+`predictions`: a list of predictions to score. Each prediction is a string that contains natural language and LaTeX.
+
+ `references`: list of reference for each prediction. Each reference is a string that contains natural language and LaTeX.
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:15]: Nit
```suggestion
`references`: list of reference for each prediction. Each reference is a string that contains natural language and LaTeX.
```

File: metrics/competition_math/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+N.B. To be able to use Competition MATH, you need to install the `math_equivalence` dependency using `pip install git+https://github.com/hendrycks/math.git`. 
+
+
+## Output values
+
+This metric returns a dictionary that contains the [accuracy](https://huggingface.co/metrics/accuracy) after canonicalizing inputs, on a scale between 0.0 and 1.0.
+
+### Values from popular papers
+The [original MATH dataset paper](https://arxiv.org/pdf/2103.03874.pdf) reported accuracies ranging from 3.0% to 6.9% by different large language models.
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:34]: Better abs:
```suggestion
The [original MATH dataset paper](https://arxiv.org/abs/2103.03874) reported accuracies ranging from 3.0% to 6.9% by different large language models. 
```

File: metrics/competition_math/README.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+This metric returns a dictionary that contains the [accuracy](https://huggingface.co/metrics/accuracy) after canonicalizing inputs, on a scale between 0.0 and 1.0.
+
+### Values from popular papers
+The [original MATH dataset paper](https://arxiv.org/pdf/2103.03874.pdf) reported accuracies ranging from 3.0% to 6.9% by different large language models. 
+
+More recent progress on the dataset can be found on the [dataset leaderboard](https://paperswithcode.com/sota/math-word-problem-solving-on-math).
+
+## Examples 
+
+Maximal values (full match) :
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:40]: Nit
```suggestion
Maximal values (full match):
```

File: metrics/competition_math/README.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+`predictions`: a list of predictions to score. Each prediction is a string that contains natural language and LaTeX.
+
+ `references`: list of reference for each prediction. Each reference is a string that contains natural language and LaTeX.
+
+
+```python
+>>> from datasets import load_metric
+>>> math = load_metric("competition_math")
+>>> references=["\\frac{1}{2}"]
+>>> predictions=["1/2"]
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:22]: Style
```suggestion
>>> references = ["\\frac{1}{2}"]
>>> predictions = ["1/2"]
```

File: metrics/competition_math/README.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+## Examples 
+
+Maximal values (full match) :
+
+```python
+>>> from datasets import load_metric
+>>> math = load_metric("competition_math")
+>>> references=["\\frac{1}{2}"]
+>>> predictions=["1/2"]
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:46]: Style:
```suggestion
>>> references = ["\\frac{1}{2}"]
>>> predictions = ["1/2"]
```

File: metrics/competition_math/README.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+{'accuracy': 1.0}
+```
+
+Minimal values (no match):
+
+```python
+>>> from datasets import load_metric
+>>> math = load_metric("competition_math")
+>>> references=["\\frac{1}{2}"]
+>>> predictions=["3/4"]
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:58]: Style:
```suggestion
>>> references = ["\\frac{1}{2}"]
>>> predictions = ["3/4"]
```

File: metrics/competition_math/README.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+{'accuracy': 0.0}
+```
+
+Partial match:
+
+```python
+>>> from datasets import load_metric
+>>> math = load_metric("competition_math")
+>>> references=["\\frac{1}{2}","\\frac{3}{4}"]
+>>> predictions=["1/5", "3/4"]
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:70]: Style:
```suggestion
>>> references = ["\\frac{1}{2}","\\frac{3}{4}"]
>>> predictions = ["1/5", "3/4"]
```

File: metrics/competition_math/README.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+>>> results = math.compute(references=references, predictions=predictions)
+>>> print(results)
+{'accuracy': 0.5}
+```
+
+## Limitations and bias
+
+This metric is limited to datasets with the same format as the [Mathematics Aptitude Test of Heuristics (MATH) dataset](https://huggingface.co/datasets/competition_math), and is meant to evaluate the performance of large language models at solving mathematical problems.
+
+N.B. The MATH dataset also assigns levels of difficulty to different problems, so disagregating model performance by difficulty level (similarly to what was done in the [original paper](https://arxiv.org/pdf/2103.03874.pdf) can give a better indication of how a given model does on a given difficulty of math problem, compared to overall accuracy.
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:80]: Better abs:
```suggestion
N.B. The MATH dataset also assigns levels of difficulty to different problems, so disagregating model performance by difficulty level (similarly to what was done in the [original paper](https://arxiv.org/abs/2103.03874) can give a better indication of how a given model does on a given difficulty of math problem, compared to overall accuracy. 
```

File: metrics/competition_math/README.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    and Jacob Steinhardt},
+  journal={arXiv preprint arXiv:2103.03874},
+  year={2021}
+}
+```
+    
+## Further References 
+- [MATH dataset](https://huggingface.co/datasets/competition_math)
+- [MATH leaderboard](https://paperswithcode.com/sota/math-word-problem-solving-on-math)
+- [MATH paper](https://arxiv.org/pdf/2103.03874.pdf)
```
Comment (albertvillanova) [inline metrics/competition_math/README.md:103]: Abs:
```suggestion
- [MATH paper](https://arxiv.org/abs/2103.03874)
```

**Review by albertvillanova — APPROVED** [review]
Thank you!

**Review by lhoestq — APPROVED** [review]
Love it, thanks !

**Pull request merged**